lib/util/escapeRegExp.js

```diff
@@ -0,0 +1,43 @@
+/**
+ * Escapes RegExp special characters from a string
```

Comment:
I understand but right now what i'm doing is enabling users to use `*` as a wildcard only, the user won't be able to use a regexp in the pattern; only `*` as a wildcard. If you're suggesting dropping the regexp implementation detail altogether could you suggest how to handle cases like: `'*-foo-*'`, `'*-foo-*-bar'`, `'*-foo-*bar*-baz*'` and so on in a performant manner?